.eslintrc.js

```diff
@@ -6,38 +6,138 @@ module.exports = {
+                            },
+                        ],
+                    },
+                ],
+            },
+        },
+        {
+            files: ['*.js', '*.jsx'],
+            settings: {
+                'import/resolver': {
+                    node: {
+                        extensions: ['.js', '.website.js', '.desktop.js', '.native.js', '.ios.js', '.android.js', '.config.js', '.ts', '.tsx'],
+                    },
+                },
+            },
+            rules: {
+                'import/extensions': [
+                    'error',
+                    'ignorePackages',
+                    {
+                        js: 'never',
+                        jsx: 'never',
+                        ts: 'never',
+                        tsx: 'never',
+                    },
+                ],
+            },
+        },
+        {
+            files: ['*.ts', '*.tsx'],
+            extends: ['airbnb-typescript', 'plugin:@typescript-eslint/recommended', 'plugin:@typescript-eslint/recommended-requiring-type-checking', 'prettier'],
+            plugins: ['@typescript-eslint', 'jsdoc'],
+            parser: '@typescript-eslint/parser',
+            parserOptions: {
+                project: './tsconfig.json',
+            },
+            rules: {
+                // TODO: Remove the following rules after TypeScript migration is complete.
+                '@typescript-eslint/no-unsafe-argument': 'off',
+                '@typescript-eslint/no-unsafe-call': 'off',
+                '@typescript-eslint/no-unsafe-member-access': 'off',
+                '@typescript-eslint/no-unsafe-assignment': 'off',
+
+                '@typescript-eslint/naming-convention': [
+                    'error',
                     {
-                        name: 'react-native',
-                        importNames: ['useWindowDimensions'],
-                        message: 'Please use useWindowDimensions from src/hooks/useWindowDimensions instead',
+                        selector: 'variable',
+                        format: ['camelCase', 'UPPER_CASE', 'PascalCase'],
                     },
                     {
-                        name: 'react-native',
-                        importNames: ['TouchableOpacity', 'TouchableWithoutFeedback', 'TouchableNativeFeedback', 'TouchableHighlight'],
-                        message: 'Please use PressableWithFeedback and/or PressableWithoutFeedback from src/components/Pressable instead',
+                        selector: ['property', 'function'],
+                        format: ['camelCase', 'PascalCase'],
                     },
                     {
-                        name: 'react-native',
-                        importNames: ['StatusBar'],
-                        message: 'Please use StatusBar from src/libs/StatusBar instead',
+                        selector: ['enumMember', 'typeLike', 'typeParameter'],
```

Comment:
```suggestion
                        selector: ['enum', 'enumMember', 'typeLike', 'typeParameter'],
```

Shouldn't we add `enum` as well?

Reply:
From docs: `typeLike - matches the same as class, enum, interface, typeAlias, typeParameter`

So we don't need `enum` and `typeParameter`.